src/osemgrep/language_server/Session.mli

```diff
@@ -10,7 +10,7 @@ type t = {
   incoming : Lwt_io.input_channel;
   outgoing : Lwt_io.output_channel;
   workspace_folders : Fpath.t list;
-  documents : (Fpath.t, Semgrep_output_v1_t.cli_match list) Hashtbl.t;
+  cached_scans : (Fpath.t, Semgrep_output_v1_t.cli_match list) Hashtbl.t;
```

Comment:
You do not store also the time at which the scan was done on this file? So if a file is modified, how do you know
the cache is invalid and we should rerun semgrep for this file?

Reply:
We scan everytime the file is changed, or when its saved, and also initially when a project is opened. This will stay up to date unless a user performs a git checkout (or modifies a file with a different editor, but that's not something easy to handle). This was already an issue that creates phantom diagnostics. What we should do is setup lsp "file watchers" to detect when git HEAD changes, and then rescan, but that's something that should be in a separate PR